### PR TITLE
Support model param in de/tokenize

### PR DESCRIPTION
**Pull request by abdullahkady**

### CODE REVIEW

File: tests/sync/test_tokenize.py [hunk 1]
```diff
@@ -8,6 +9,17 @@
 
 
 class TestTokenize(unittest.TestCase):
+    @pytest.mark.skipif(True, reason="until this is available globally in the API")
```
Comment (harry-cohere) [inline tests/sync/test_tokenize.py:12]: I believe the sync tests don't use pytest. Instead you can skip the test using the unittest decorators: https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures

**Review by harry-cohere — CHANGES_REQUESTED** [review]

File: tests/sync/test_tokenize.py (hunk 1 above)
Comment (abdullahkady) [reply tests/sync/test_tokenize.py:12]: @harry-cohere we still use pytest as our runner, so the decorators will take effect normally:
https://github.com/cohere-ai/cohere-python/blob/main/.github/workflows/test.yaml#L29-L32

File: tests/async/test_async_tokenize.py [hunk 2]
```diff
@@ -25,14 +38,27 @@ async def test_invalid_text(async_client):
...
-    detokenizeds = await async_client.batch_detokenize([[10104, 12221, 974, 514, 34]] * 3)
+    detokenizeds = await async_client.batch_detokenize([[10104, 12221, 974, 514, 34]] * 3, model="xlarge")
     assert detokenizeds == [Detokenization("detokenize me!")] * 3
 
     detokenized = await async_client.detokenize([])
     assert detokenized == Detokenization("")
+
+
+@pytest.mark.asyncio
+@pytest.mark.skipif(True, reason="until this is available globally in the API")
```
[the remark this answers is not in the document]
Comment (abdullahkady) [reply tests/async/test_async_tokenize.py]: just fyi .. these comments shouldn't be relevant cuz the plan is to remove these skip lines within the PR before merging anyway

[the remark this answers is not in the document]
Comment (abdullahkady) [reply tests/async/test_async_tokenize.py]: I'm just using it now to run CI 😅

**Review by harry-cohere — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 4.6.2
```
Comment (sanderland) [inline CHANGELOG.md:3]: would do 4.7, this is not a bugfix but a new feature

File: cohere/client.py [hunk 4]
```diff
@@ -432,53 +432,61 @@ def summarize(
...
-    def tokenize(self, text: str) -> Tokens:
+    def tokenize(self, text: str, model: Optional[str] = None) -> Tokens:
         """Returns a Tokens object of the provided text, see https://docs.cohere.ai/reference/tokenize for advanced usage.
 
         Args:
             text (str): Text to summarize.
+            model (str): An optional model name that will ensure that the tokenization uses the tokenizer used by that model, which can be critical for counting tokens properly.
         """
         json_body = {"text": text}
+        if model is not None:
```
Comment (sanderland) [inline cohere/client.py:458]: does the API require this? we usually just pass the None on

File: tests/async/test_async_tokenize.py [hunk 5]
```diff
@@ -25,14 +38,27 @@ async def test_invalid_text(async_client):
 
 @pytest.mark.asyncio
 async def test_detokenize(async_client):
-    detokenized = await async_client.detokenize([10104, 12221, 974, 514, 34])
+    detokenized = await async_client.detokenize([10104, 12221, 974, 514, 34], model="xlarge")
```
Comment (sanderland) [inline tests/async/test_async_tokenize.py:41]: we call this 'base' now right

**Review by sanderland — APPROVED** [review]

**Review by mkozakov — APPROVED** [review]

**Pull request merged**